clarity-ext-scripts/clarity_ext_scripts/covid/knm_sminet_service.py

```diff
@@ -0,0 +1,157 @@
+
+        return SampleInfo(status=1,
+                          sample_id=sample.org_referral_code,
+                          sample_date_arrival=sample.date_arrival,
+                          sample_date_referral=sample_date_referral,
+                          sample_material="Svalg",
+                          sample_free_text_referral="Anamnes: Personalprov")
+
+    def create_referring_clinic(self, provider):
+        """Creates a referring clinic object from KNM data"""
+        referring_clinic_name = provider.patient["managingOrganization"]["display"]
+        referring_clinic_county = self.get_county_from_organization(
+            provider.organization)
+        referring_doctor = provider.service_request["resource"]["requester"]["display"]
+
+        # TODO: We don't have an address for the referring clinic
+        return ReferringClinic(referring_clinic_name, "",
+                               referring_clinic_county, Doctor(referring_doctor))
+
+
+    def map_gender(self, knm_gender):
+        """Returns gender in the format required by SmiNet"""
+        pass
+
+    def create_patient(self, provider):
+        """Creates a Patient object from KNM data"""
+
+        def patient_identifier():
+            try:
+                patient_identifier = provider.patient["identifier"]
+            except KeyError:
+                raise IntegrationError(
+                    "Missing field 'identifier' on the patient resource for {}".format(provider))
+
+            if len(patient_identifier) == 0:
+                raise IntegrationError(
+                    "Field 'identifier' is empty on patient resource for {}".format(provider))
+
+            try:
+                return patient_identifier[0]["value"]
+            except:
+                raise IntegrationError(
+                    "First entry in 'identifier' doesn't have a value key for {}".format(provider))
+
+        def patient_gender():
+            gender = provider.patient["gender"]
+            try:
+                return self.MAP_GENDER_FROM_KNM_TO_SMINET[gender]
+            except KeyError:
+                raise IntegrationError(
+                    "Unexpected value for key 'gender' ({}) for {}".format(gender, provider))
+
+        def patient_name():
+            """
+            The name of the patient. Returns an empty string if we don't have the expected data
+            """
+            name = provider.patient["name"]
+
+            if len(name) == 0:
+                return ""
+
+            name = name[0]
+
+            try:
+                return name["text"]
+            except KeyError:
+                return ""
+
+        return Patient(patient_identifier(),
+                       patient_gender(),
+                       patient_name(),
+                       None)
+
+    def get_sminet_export(self, sample):
+        """
+        Returns a SmiNetLabExport from a sample 
+
+        :sample: A KNMSampleAccessor object
+        """
```

Comment:
I don't think it necessarily needs to change now, but think that it would be good if the hard coded parts in this method, like the doctors name, etc would be moved to the extension using it. I'd argue that that is place to put it because it is the part of the code that future developers are most likely to interact with, and thus be aware of it being hard coded. As I said no need to change it now, but just wanted to went my thoughts on the subject.

Reply:
Right, I intend to pull up all the constants.